### PR TITLE
implement mmer index for 19% faster queries though 15 second slower DB loading

### PR DESCRIPTION
**Pull request by boris-dimitrov**
todo:  prebuild the mmer index or load it in parallel (this will work)

### CODE REVIEW

File: src/gt_pro.cpp [hunk 1]
```diff
@@ -104,21 +112,14 @@ long chrono_time() {
 	return duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count();
 }
 
-using LmerRange = tuple<uint64_t, uint64_t>;
-
-void kmer_lookup(LmerRange* lmer_indx, vector<uint32_t>& mmers, vector<uint64_t>& snps, int channel, char* in_path, char* o_name){
-	uint32_t lsb = 1;
-	uint32_t b_mask = (lsb << bpb) - lsb;
+using LmerRange = tuple<uint32_t, uint8_t>;
```
Comment (boris-dimitrov) [inline src/gt_pro.cpp]: We might want to do something here to handle the case of a bigger DB.   So far both DBs fit under 32 bits, but the bigger DB just barely, and we should probably add at least an assert.

Comment (boris-dimitrov) [reply src/gt_pro.cpp]: I should also add a comment that the tagset DB distribution of LmerRange length is median 2, p99 8, and max 174; and assert on length > 255.

Comment (zjshi) [reply src/gt_pro.cpp]: Awesome changes, thanks!

Comment (zjshi) [reply src/gt_pro.cpp]: That is my original concern compelling me to use something like 64bit, but I could be just paranoid. If it also sounds a legit concern to you to scale up, I would probably make it changable by taking a silent argument indicating the rough total number of kmers in the database and deciding which int type to use based on that. Does it sound good to you?

Comment (boris-dimitrov) [reply src/gt_pro.cpp]: Yes, I agree.   I'll make that happen.

Comment (zjshi) [reply src/gt_pro.cpp]: Awesome work, Boris! I am going to do merging shortly, thanks!

**Pull request merged**